OpenStack: Automate UPI with Kuryr

MaysaMacedo opened the pull request:
Update ansile playbooks to support setting Kuryr as networking
type for a User Provided installation.

MaysaMacedo commented:
/label platform/openstack

luis5tb commented:
@MaysaMacedo: I think in the kuryr case we also need to create the service net/subnets, as well as the subnet pool for the pods

MaysaMacedo commented:
> @MaysaMacedo: I think in the kuryr case we also need to create the service net/subnets, as well as the subnet pool for the pods

@luis5tb Thanks. 

The service net/subnets and the subnet pool are created at the Kuryr Bootstrap on CNO. But, we're missing the deletion of the subnet pool, gonna tackle that.

luis5tb commented:
> > @MaysaMacedo: I think in the kuryr case we also need to create the service net/subnets, as well as the subnet pool for the pods
> 
> @luis5tb Thanks.
> 
> The service net/subnets and the subnet pool are created at the Kuryr Bootstrap on CNO. But, we're missing the deletion of the subnet pool, gonna tackle that.

We should have the option for the user to pass them too. Note in non-kuryr case is different as that is internal to OCP installation. But in the Kuryr case, that is on the infra side (aka OpenStack side).

MaysaMacedo commented:
/assign @tomassedovic
/assign @pierreprinetti

cc @luis5tb @dulek

MaysaMacedo commented:
/retest

MaysaMacedo commented:
/retest

MaysaMacedo commented:
/retest

MaysaMacedo commented:
/test e2e-libvirt

pierreprinetti commented:
/assign
/approve

Tested with OpenShiftSDN, works beautifully. Great work!

Do you think you can separate the part where you add tags from the rest of the patch, before I LGTM?

pierreprinetti commented on upi/openstack/inventory.yaml:
```diff
@@ -15,6 +15,9 @@ all:
       os_ingress_fip: '203.0.113.19'
 
+      os_region_name: ''
+      os_networking_type: 'OpenshiftSDN'
```
Can you document this field? It's clear in code that the relevant value is `Kuryr`, however I think it would be nice to add a comment here or in the README

pierreprinetti commented on upi/openstack/down-06_load-balancers.yaml:
```diff
@@ -0,0 +1,16 @@
+  gather_facts: no
+
+  tasks:
+  - name: 'Remove the cluster load balancers'
```
This playbook does not have a corresponding "create" counterpart. Can you document its purpose in the README?

MaysaMacedo commented:
> /assign
> /approve
> 
> Tested with both `OpenShiftSDN` and `Kuryr` as `os_networking_type`, it successfully installed in both cases. Great work!
> 
> However, `down-06_load-balancers.yaml` failed with `Unable to import os_delete_lb_resources due to invalid syntax`.
> 
> Do you think you can split the PR to separate the part where you add tags from the rest?

@pierreprinetti Thanks for your review! I just updated with your suggestions, and fixed the issue with the lb playbook.
Regarding the split of the PR, I can do that, but installation with Kuryr and without tagging resources will not succeed. My idea was to combine everything needed for a successful installation with Kuryr in one PR. What do you think?

pierreprinetti commented:
> My idea was to combine everything needed for a successful installation with Kuryr in one PR.

Makes sense. What about two separate commits in the same PR?

MaysaMacedo commented:
> > My idea was to combine everything needed for a successful installation with Kuryr in one PR.
> 
> Makes sense. What about two separate commits in the same PR?

Good idea. I will do that!

MaysaMacedo commented:
/retest

MaysaMacedo commented:
/retest

MaysaMacedo commented:
/retest

MaysaMacedo commented:
/retest

MaysaMacedo commented:
/retest

MaysaMacedo commented:
/retest

MaysaMacedo commented:
/test e2e-libvirt

pierreprinetti commented:
/lgtm

pierreprinetti commented:
/hold
/test all

pierreprinetti commented:
/test e2e-openstack

pierreprinetti commented:
/hold cancel

ok, the CI looks better now. :crossed_fingers:
